### PR TITLE
Add comprehensive testing of the janestreet profile

### PR DESCRIPTION
**Pull request by dvulakh**
This repo includes a large number of tests that stress many shapes of input code. However, most of these tests do not currently exercise the `janestreet` profile.

This PR extends the test suite in `test/passing/tests` with `*.js-ref` and `*.js-err` files. They are analagous to `*.ref` and `*.err` files, except that
- They are formatted with `--profile=janestreet`
- They ignore extra command-line arguments in `*.opts` files

To ensure comprehensiveness, the build rules require that every test for the default profile have either a corresponding `*.js-ref` file, or a corresponding `*.why-no-js` file (that file should contain a short message describing why the test is not relevant to the janestreet profile).

This PR also merges `js_source.ml` and `source.ml`, since `source.ml.js-ref` can now test what was previously stressed in `js_source.ml.ref`. One structure item formerly in `js_source.ml` is split out into `comment_long_js_only.ml`, because the `ocamlformat` profile does not stabilize on it.

This PR should be reviewed together with the internal re-import, which shows that the tests added here agree with the (more limited) internal testing we have been applying until now.

### CODE REVIEW

File: test/passing/gen/gen.ml [hunk 1]
```diff
@@ -93,6 +97,49 @@ let cmd should_fail args =
...
+  in
+  let err_msg = Printf.sprintf err_msg test_name in
+  Printf.sprintf
+    {|
+(rule
+ (alias runtest)
+ (enabled_if (<> %%{os_type} Win32))
+ (package ocamlformat)
+ (action (system "if [%s -f tests/%s.why-no-js ]; then echo '%s'; exit 1; fi")))
+|}
```
Comment (mdelvecchio-jsc) [inline test/passing/gen/gen.ml:146]: We know during the running of `gen.ml` whether or not both/neither are present; lets only print out a rule like this if we know it's going to fail.

**Review by mdelvecchio-jsc — COMMENTED** [review]
Looks good other than the suggestion I left.

File: test/passing/gen/gen.ml (hunk 1 above)
Comment (dvulakh) [reply test/passing/gen/gen.ml:146]: Good idea. Done.

**Review by mdelvecchio-jsc — APPROVED** [review]
lgtm

**Pull request merged**